### PR TITLE
Feature/oep16 providers rewards

### PR DESCRIPTION
**Pull request by 0x3bfc**

### CODE REVIEW

File: 16/README.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+# 3. Actors and Technical Components
+
+* SOURCERS - Provide access to assets and/or services
+* PROVIDERS - Provide storage in IPFS network and serve the download request
+* CONSUMERS - Want to get access to assets and/or services
+* MARKETPLACES - Store the DDO (including metadata) associated with the assets and/or services
+
+The following technical components are involved with the publishing flow or the consumption flow:
+
+* MARKETPLACE - Exposes a web interface allowing users to publish and purchase assets. Also facilitates the discovery of assets. An example MARKETPLACE web interface is [Pleuston](https://github.com/oceanprotocol/pleuston).
```
Comment (0x3bfc) [inline 16/README.md]: Use common as a reference not Plauston, AFAIK, it is used to be a marketplace for priced data and the reward meant to be for free/public datasets.

File: 16/README.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+* [SQUID](https://github.com/oceanprotocol/dev-ocean/blob/master/doc/architecture/squid.md) - Software library encapsulating the Ocean Protocol business logic. It's used to interact with all the components & APIs of the system. It's currently implemented in the following packages:
+  * [squid-js](https://github.com/oceanprotocol/squid-js) - JavaScript version of SQUID to be integrated with front-end applications.
```
Comment (0x3bfc) [inline 16/README.md]: We don't have to list all the components in texts. just refer to them through link. we need to keep the OEP short and focused on the changes.

The same for keeper, secretstore, and brizo.

File: 16/README.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+* [KEEPER CONTRACTS](https://github.com/oceanprotocol/keeper-contracts) - Provide the Service Agreement business logic.
+* [SECRET STORE](https://github.com/oceanprotocol/parity-ethereum) - Included as part of the Parity Ethereum client. Allows the PUBLISHER to encrypt the asset URL. Integrates with the SA to authorize (on-chain) the decryption of the asset URL by the CONSUMER.
+* [BRIZO](https://github.com/oceanprotocol/brizo) - Microservice to be executed by PUBLISHERS. It exposes an HTTP REST API permitting access to PUBLISHER assets or additional services such as computation.
+* [AQUARIUS](https://github.com/oceanprotocol/aquarius) - Microservice to be executed by MARKETPLACES. Facilitates creating, updating, deleting and searching the asset metadata registered by the PUBLISHERS. This metadata is included as part of a DDO (see [OEP-7](../7/README.md) and [OEP-8](../8/README.md)) and also includes the services associated with the asset (consumption, computation, etc.).
+
+# 4. Work Flows
+
+This section describes the network reward distribution workflow in details.
```
Comment (0x3bfc) [inline 16/README.md]: This is a good one 👍

File: 16/README.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+
+# 4. Work Flows
+
+This section describes the network reward distribution workflow in details. 
+
+## 4.1 Overall Architecture
+
+From the high level, the workflow is illustrated as below:
+
+<img src="img/distribution.jpg" />
```
Comment (0x3bfc) [inline 16/README.md]: This is a good one 👍

File: 16/README.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+	```
+	
+	The parameters are:
+	
+	* `ipfs_id`: the IPFS node ID that looks like `QmVrZp6L...`
+	* `public_key`: the RSA public key that will be hashed into IPFS node ID.
+
+
+## 4.3 Token Mining
+
```
Comment (0x3bfc) [inline 16/README.md]: The exciting stuff should be here 👯‍♂

File: 16/README.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+	* The actual reward amount is calculated according to formula as described in [V3 Network Backend Specs](https://docs.google.com/document/d/1zgf-rVW0UrAokJ2GZr_k4RKrcqgaP18Yj0aqMK16rh8/)
+	* Ocean distributes the correct amount of network rewards to chosen winner. 
+	* repeat the process every period of time.
+
+To implement this workflow, some other modules are needed to be implemented or integrated, such as IFPS node registration, random number generator/Oracle network service and etc.
+
+
+## 4.2 IPFS node registration
+
+Since the IPFS network is separate from Ocean network, it is required for IPFS nodes to register their Ethereum wallet in Ocean to receive reward tokens.
```
Comment (0x3bfc) [inline 16/README.md]: I have discussed this @alexcos20 and we are going to move the IPFS stuff to this OEP-17 (  https://github.com/oceanprotocol/OEPs/pull/new/feature/OEP-17-miner) 
@gongf05 Feel free to update.

File: 16/README.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+
+## 4.1 Overall Architecture
+
+From the high level, the workflow is illustrated as below:
+
+<img src="img/distribution.jpg" />
+
+
+* **Token Mining**:
+	* every period of time, Ocean will mine new tokens as the network rewards, which will be distributed to providers as the incentive.
```
Comment (H34D) [inline 16/README.md]: `Ocean will mine new` -> `Ocean will mint new`

File: 16/README.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+editor: Alex Coseru <alex@oceanprotocol.com> 
+		 Fang Gong <fang@oceanprotocol.com>
+		 Ahmed Ali <ahmed@oceanprotocol.com>,
+		 Sebastian Gerske <sebastian@oceanprotocol.com>
+```
+
+
+## 1. Abstract
+
+It is critical for Ocean network to motivate data providers to add more valuable and releavent data commons to our network. As such, the total value of the network as well as Ocean token price can increase over the time. To do so, Ocean network has reserved a partial of total token supply as the network rewards that will be distributed to providers of data commons in the Ocean network.
```
Comment (H34D) [inline 16/README.md]: `rewards that will be distributed to providers of data commons in the Ocean network`

also data that is not commons should be rewarded or?

File: 16/README.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+
+From the high level, the workflow is illustrated as below:
+
+<img src="img/distribution.jpg" />
+
+
+* **Token Mining**:
+	* every period of time, Ocean will mine new tokens as the network rewards, which will be distributed to providers as the incentive.
+* **Token Distribution**:
+	* Ocean queries the provider list from IPFS for all downloaded dataset;
```
Comment (H34D) [inline 16/README.md]: why not using SEAs as base and use ipfs to verify? This would lead to a solution that works also in non-ipfs cases.

**Comment by trentmc** [comment]
Obsolete in light of updated roadmap (datatokens, oceandao/rewards).

**Pull request closed without merging**